xcm/pallet-xcm-benchmarks/src/generic/benchmarking.rs

```diff
@@ -0,0 +1,311 @@
+
+		let instruction = Instruction::<XcmCallOf<T>>::BuyExecution {
+			fees: (fee_asset, 100_000_000).into(), // should be something inside of holding
+			weight_limit: WeightLimit::Unlimited,
+		};
+
+		let xcm = Xcm(vec![instruction]);
+	} : {
+		executor.execute(xcm)?;
+	} verify {
+
+	}
+
+	// Worst case scenario for this benchmark is a large number of assets to
+	// filter through the reserve.
+	reserve_asset_deposited {
+		let mut executor = new_executor::<T>(Default::default());
+		// TODO real max assets
+		let assets = (0..MAX_ASSETS).map(|i| MultiAsset {
+			id: Abstract(i.encode()),
+			fun: Fungible(i as u128),
+
+		}).collect::<vec::Vec<_>>();
+		let multiassets: MultiAssets = assets.into();
+
+		let instruction = Instruction::ReserveAssetDeposited(multiassets.clone());
+		let xcm = Xcm(vec![instruction]);
+	}: {
+		executor.execute(xcm).map_err(|_| BenchmarkError::Skip)?;
+	} verify {
+		assert_eq!(executor.holding, multiassets.into());
+	}
+
+	query_response {
+		let mut executor = new_executor::<T>(Default::default());
+		let (query_id, response) = T::worst_case_response();
+		let max_weight = u64::MAX;
+		let instruction = Instruction::QueryResponse { query_id, response, max_weight };
+		let xcm = Xcm(vec![instruction]);
+	}: {
+		executor.execute(xcm)?;
+	} verify {
+		// The assert above is enough to show this XCM succeeded
+	}
+
+	// We don't care about the call itself, since that is accounted for in the weight parameter
+	// and included in the final weight calculation. So this is just the overhead of submitting
+	// a noop call.
+	transact {
+		let origin = T::transact_origin().ok_or(BenchmarkError::Skip)?;
+		let mut executor = new_executor::<T>(origin);
+		let noop_call: <T as Config>::Call = frame_system::Call::remark_with_event {
+			remark: Default::default()
+		}.into();
+		let double_encoded_noop_call: DoubleEncoded<_> = noop_call.encode().into();
+
+		let instruction = Instruction::Transact {
+			origin_type: OriginKind::SovereignAccount,
+			require_weight_at_most: noop_call.get_dispatch_info().weight,
+			call: double_encoded_noop_call,
+		};
+		let xcm = Xcm(vec![instruction]);
+
+		let num_events = frame_system::Pallet::<T>::events().len();
+	}: {
+		executor.execute(xcm)?;
+	} verify {
+		// TODO make better assertion?
+		let num_events2 = frame_system::Pallet::<T>::events().len();
+		assert_eq!(num_events + 1, num_events2);
+	}
+
+	refund_surplus {
+		let mut executor = new_executor::<T>(Default::default());
+		executor.total_surplus = 1337;
+		executor.total_refunded = 0;
+
+		let instruction = Instruction::<XcmCallOf<T>>::RefundSurplus;
+		let xcm = Xcm(vec![instruction]);
+	} : {
+		let result = executor.execute(xcm)?;
+	} verify {
+		assert_eq!(executor.total_surplus, 1337);
+		assert_eq!(executor.total_refunded, 1337);
+	}
+
+	set_error_handler {
+		let mut executor = new_executor::<T>(Default::default());
+		let instruction = Instruction::<XcmCallOf<T>>::SetErrorHandler(Xcm(vec![]));
```

Comment:
It looks to me that the worst complexity of the `SetErrorHandler` instruction is dependent upon the Weigher trying to weight the error handler's XCM instructions, and so having a 0 length vec is the best case.

Reply:
Weight calculation itself should be trivial to calculate (minimal operations, no storage reads), so practically speaking, this should be fine I think.